### PR TITLE
Add and apply scalafmt

### PR DESCRIPTION
**Pull request by mdedetrich**
This PR adds scalafmt, as well as a github action check as well as applying it to the codebase.

When merging the PR, either do a merge commit or a rebase and merge but not squash and merge as I want to add the `Apply scalafmt` commit to git blame ignore revs.

I used a scalafmt configuration that is common within sbt-plugins

### CODE REVIEW

File: .scalafmt.conf [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      { regex = "Term.ApplyInfix" }
+    ]
+  },
+  {
+    code   = "%%"
+    owners = [
+      { regex = "Term.ApplyInfix" }
+    ]
+  }
+]
```
Comment (raboof) [inline .scalafmt.conf:26]: do we really want this? It's nonstandard, IMHO it doesn't help readability that much, and makes diffs bigger than necessary...

**Review by raboof — APPROVED** [review]
I'm OK with this (though I have my doubts about aligning `%`/`%%`), seems helpful to get this merged now that there's few open PRs.

File: .scalafmt.conf (hunk 1 above)
Comment (mdedetrich) [reply .scalafmt.conf:26]: I mean not neccessarily, its just that as I said most sbt plugins already use this config (I happen to maintain quite a few).

Do you feel strongly about this? I personally find it easier to read but in cases like this I defer to consistency

**Review by eed3si9n — APPROVED** [review]
LGTM

File: .scalafmt.conf (hunk 1 above)
Comment (eed3si9n) [reply .scalafmt.conf:26]: I'd say do whatever makes you happy to contribute to open source :)

Comment (mdedetrich) [reply .scalafmt.conf:26]: @raboof I will merge as is so we can progress forward, if you feel strongly about this change we can always update it later

Comment (raboof) [reply .scalafmt.conf:26]: > Do you feel strongly about this?

No, otherwise I wouldn't have approved the PR :)

**Pull request merged**